src/credentials/attestation_verifier/TestDACRevocationDelegateImpl.cpp

```diff
@@ -30,13 +31,50 @@ namespace chip {
 namespace Credentials {
 
 namespace {
+
+static constexpr uint32_t kMaxIssuerBase64Len = BASE64_ENCODED_LEN(kMaxCertificateDistinguishedNameLength);
+
 CHIP_ERROR BytesToHexStr(const ByteSpan & bytes, MutableCharSpan & outHexStr)
 {
     Encoding::HexFlags flags = Encoding::HexFlags::kUppercase;
     ReturnErrorOnFailure(BytesToHex(bytes.data(), bytes.size(), outHexStr.data(), outHexStr.size(), flags));
     outHexStr.reduce_size(2 * bytes.size());
     return CHIP_NO_ERROR;
 }
+
+CHIP_ERROR X509_PemToDer(const std::string & pemCert, MutableByteSpan & derCert)
+{
+    std::string beginMarker = "-----BEGIN CERTIFICATE-----";
+    std::string endMarker   = "-----END CERTIFICATE-----";
+
+    std::size_t beginPos = pemCert.find(beginMarker);
+    VerifyOrReturnError(beginPos != std::string::npos, CHIP_ERROR_INVALID_ARGUMENT);
+
+    std::size_t endPos = pemCert.find(endMarker);
+    VerifyOrReturnError(endPos != std::string::npos, CHIP_ERROR_INVALID_ARGUMENT);
+
+    // Extract content between markers
+    std::string plainB64Str = pemCert.substr(beginPos + beginMarker.length(), endPos - (beginPos + beginMarker.length()));
```

Comment:
What happens if `endPos < beginPos` here?

Reply:
added that check